File: 1137-n-th-tribonacci-number/1137-n-th-tribonacci-number.java
class Solution {
    int cache[] = new int[100];
    public int tribonacci(int n) {
        if (n<=1) {
            return n;
        }
        if (n==2) {
            return 1;
        }
        
        if (cache[n-1] == 0) {
            cache[n-1] = tribonacci(n-1);
        }
         if (cache[n-3] == 0) {
            cache[n-3] = tribonacci(n-3);
        }
         if (cache[n-2] == 0) {
            cache[n-2] = tribonacci(n-2);
        }
        return cache[n-1]+cache[n-2]+cache[n-3];
    }
}

File: 1137-n-th-tribonacci-number/1137-n-th-tribonacci-number.cpp
class Solution {
public:
    int tribonacci(int n) {
        int arr[3];
        arr[0] = 0;
        arr[1] = 1;
        arr[2] = 1;
        if(n < 3)
            return arr[n];
        for(int i = 3; i <= n; i++)
            arr[i%3] = arr[0] + arr[1] + arr[2];
        return arr[n%3];
    }
};